src/GraphQLParser/AST/GraphQLName.cs

```diff
@@ -11,6 +11,21 @@ public class GraphQLName : ASTNode
     /// <inheritdoc/>
     public override ASTNodeKind Kind => ASTNodeKind.Name;
 
+    /// <summary>
+    /// Creates a new instance (with empty value).
+    /// </summary>
+    public GraphQLName()
+    {
+    }
+
+    /// <summary>
+    /// Creates a new instance with the specified value.
+    /// </summary>
+    public GraphQLName(ROM value)
```

Comment:
> These changes will allow us to work with ROM more widely/freely in the main project. Many places that use `string` can be changed to `ROM` without performance drop (I think).

This will need to be verified.  If it is stored as ROM in the main project, and then converted to a string more than once, it is a guaranteed performance drop.  If it's stored as a ROM and converted to a string once, then it's the same.  Only if it's stored as a ROM and unused would there be a performance benefit.  I already tried using ROMs throughout the main project and it did not go well.  I certainly would not use ROMs in the main project 'just because we can', so if there is no performance difference, we should leave it as strings.

Reply:
ROM -> string is explicit conversion (as designed) as it _may_ allocate string.
string -> ROM is implicit as it never allocate.

I added code to compare ROM vs ROM when ROM _may be_ backed by string. 

> Only if it's stored as a ROM and unused would there be a performance benefit. I already tried using ROMs throughout the main project and it did not go well. I certainly would not use ROMs in the main project 'just because we can', so if there is no performance difference, we should leave it as strings.

Will see.

Reply:
I still do not know what happens.